src/jwplayer/utils/useAdComplete.ts

```diff
@@ -25,17 +25,21 @@ export default function useAdComplete(): boolean {
 	}, []);
 
 	const waitForAdEngine = () => {
-		// to prevent prettier reformatting this line and then spitting out errors
-		// prettier-ignore
-		const adEngineConfigured$ = communicationService.action$.pipe(ofType('[AdEngine] Configured'), first());
-		const adEngineTimeout = 2000;
-		const adEngineTimeout$ = timer(adEngineTimeout);
+		const adEngineTimeout$ = new Promise((resolve) => {
+			setTimeout(resolve, 2000, 'timeout');
+		});
+		const adEngineConfigured$ = new Promise((resolve) => {
+			communicationService.on('[AdEngine] Configured', resolve);
+		});
 
-		return race(adEngineConfigured$, adEngineTimeout$).toPromise();
+		return Promise.race([adEngineConfigured$, adEngineTimeout$]);
 	};
 
 	const listenSetupJWPlayer = (callback) => {
-		communicationService.action$.pipe(ofType('[Ad Engine] Setup JWPlayer'), first()).subscribe(callback);
+		communicationService.on('[Ad Engine] Setup JWPlayer', () => {
+			callback();
+		});
+		console.log(' ---- [Ad Engine] Setup JWPlayer ---- ');
```

Comment:
Similarly to the comment above: let's replace it for now with `console.debug()`. Also, I wonder if it shouldn't be in the callback in `communicationService.on()` above 🤔 It depends what you wanted to signal with the log -- the fact we hooked to the event or the fact the event happened and we reacted on it calling the callback?